### PR TITLE
60 add support lmfit differential evolution

### LINKED ISSUE #60: Add support LMFit differential_evolution
The LMFit differential_evolution is needed in the tutorials for EasyReflectometry

### PR DESCRIPTION
**Pull request by andped10**
The  LMFit differential_evolution method is needed in some tutorials for EasyReflectometry

### CODE REVIEW

File: src/easyscience/fitting/available_minimizers.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import warnings
+from enum import Enum
+from enum import auto
+
+import pkg_resources
+
+installed_packages = {pkg.key for pkg in pkg_resources.working_set}
```
Comment (rozyczko) [inline src/easyscience/fitting/available_minimizers.py:7]: `pkg_resources` seenms to be deprecated.
```
>>> import pkg_resources
<stdin>:1: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
```

Consider using
```python
import importlib.metadata
installed_packages = [x.name for x in importlib.metadata.distributions()]
```

File: src/easyscience/fitting/minimizers/minimizer_base.py [hunk 2]
```diff
@@ -122,6 +122,21 @@ def evaluate(self, x: np.ndarray, minimizer_parameters: dict[str, float] = None,
...
+            method['method'] = self._method
+
+        # Set / overwrite if method was passed
+        if passed_method is not None:
+            if passed_method in self.supported_methods():
+                method['method'] = passed_method
+            else:
+                raise FitError(f'Method {passed_method} not available in {self.__class__}')
+        return method
+
```
Comment (rozyczko) [inline src/easyscience/fitting/minimizers/minimizer_base.py:135]: How about removing `else` and streamlining this a bit with premature returns?

```python
def _get_method_dict(self, passed_method: Optional[str] = None) -> dict[str, str]:
    if passed_method is not None:
        if passed_method not in self.supported_methods():
            raise FitError(f'Method {passed_method} not available in {self.__class__}')
        return {'method': passed_method}

    if self._method is not None:
        return {'method': self._method}

    return {}

```

**Review by rozyczko — APPROVED** [review]
minor issues only.

File: src/easyscience/fitting/minimizers/minimizer_base.py (hunk 2 above)
Comment (andped10) [reply src/easyscience/fitting/minimizers/minimizer_base.py:135]: Thanks, done

File: src/easyscience/fitting/available_minimizers.py (hunk 1 above)
Comment (andped10) [reply src/easyscience/fitting/available_minimizers.py:7]: thanks, done

Comment (andped10) [reply src/easyscience/fitting/available_minimizers.py:7]: Reverted as `importlib.metadata.distributions` is not supported in 3.9

**Pull request merged**